### PR TITLE
Upgrade to swagger2

### LINKED ISSUE #884: Upgrade Dropwizard to version 4.x
Warning: migration to swagger V3 needed
https://www.david-merrick.com/2017/11/15/useful-regexes-for-transitioning-swagger-2-0-to-3-0-annotations/

### PR DESCRIPTION
**Pull request by alainbodiguel**
Fix #884

### CODE REVIEW

File: arlas-rest/src/main/java/io/arlas/server/rest/explore/describe/DescribeCollectionRESTService.java [hunk 1]
```diff
@@ -51,39 +54,45 @@ public DescribeCollectionRESTService(ExploreService exploreService) {
...
     @Consumes(UTF8JSON)
-    @ApiOperation(value = "Describe", produces = UTF8JSON, notes = "Describe the structure and the content of the given collection. ", consumes = UTF8JSON, response = CollectionReferenceDescription.class)
-    @ApiResponses(value = {@ApiResponse(code = 200, message = "Successful operation", response = CollectionReferenceDescription.class, responseContainer = "CollectionReferenceDescription"),
-            @ApiResponse(code = 500, message = "Arlas Server Error.", response = Error.class),
-            @ApiResponse(code = 403, message = "Collection not authorized.", response = Error.class),
-            @ApiResponse(code = 400, message = "Bad request.", response = Error.class)})
+    @Operation(summary = "Describe",  description = "Describe the structure and the content of the given collection.")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successful operation",
+                    content = @Content(array = @ArraySchema(schema = @Schema(implementation = CollectionReferenceDescription.class)))),
```
Comment (MohamedHamouGisaia) [inline arlas-rest/src/main/java/io/arlas/server/rest/explore/describe/DescribeCollectionRESTService.java:60]: it shoudn't return an array but onl an object

File: arlas-rest/src/main/java/io/arlas/server/rest/explore/search/SearchRESTService.java [hunk 2]
```diff
@@ -59,121 +64,127 @@ public SearchRESTService(ExploreService exploreService) {
...
             // --------------------------------------------------------
 
-            @ApiParam(name = "include",
-                    value = Documentation.PROJECTION_PARAM_INCLUDE,
-                    allowMultiple = true,
-                    defaultValue = "*")
+            @Parameter(name = "include",
+                    description = Documentation.PROJECTION_PARAM_INCLUDE,
+                    style = ParameterStyle.FORM,
+                    explode = Explode.TRUE,
```
Comment (MohamedHamouGisaia) [inline arlas-rest/src/main/java/io/arlas/server/rest/explore/search/SearchRESTService.java:140]: if we put explode to true, I think we then need to accept a List<string> include, i think we should not explode even though we used to say before that :allowMultiple = true => it was a mistake I'd say

File: arlas-rest/src/main/java/io/arlas/server/rest/explore/search/SearchRESTService.java [hunk 3]
```diff
@@ -59,121 +64,127 @@ public SearchRESTService(ExploreService exploreService) {
...
 
-            @ApiParam(name = "exclude",
-                    value = Documentation.PROJECTION_PARAM_EXCLUDE,
-                    allowMultiple = true,
-                    defaultValue = "")
+            @Parameter(name = "exclude",
+                    description = Documentation.PROJECTION_PARAM_EXCLUDE,
+                    style = ParameterStyle.FORM,
+                    explode = Explode.TRUE,
+                    schema = @Schema(defaultValue = ""))
```
Comment (MohamedHamouGisaia) [inline arlas-rest/src/main/java/io/arlas/server/rest/explore/search/SearchRESTService.java:148]: same remark for include

File: arlas-rest/src/main/java/io/arlas/server/rest/explore/aggregate/GeoAggregateRESTService.java [hunk 4]
```diff
@@ -72,74 +77,83 @@ public GeoAggregateRESTService(ExploreService exploreService) {
...
     @Produces(UTF8JSON)
     @Consumes(UTF8JSON)
-    @ApiOperation(value = "GeoAggregate", produces = UTF8JSON, notes = Documentation.GEOAGGREGATION_OPERATION, consumes = UTF8JSON, response = FeatureCollection.class)
-    @ApiResponses(value = {@ApiResponse(code = 200, message = "Successful operation", response = FeatureCollection.class, responseContainer = "FeatureCollection"),
-            @ApiResponse(code = 500, message = "Arlas Server Error.", response = Error.class), @ApiResponse(code = 400, message = "Bad request.", response = Error.class),
-            @ApiResponse(code = 501, message = "Not implemented functionality.", response = Error.class)})
+    @Operation(summary = "GeoAggregate",  description = Documentation.GEOAGGREGATION_OPERATION)
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successful operation",
+                    content = @Content(array = @ArraySchema(schema = @Schema(implementation = FeatureCollection.class)))),
```
Comment (MohamedHamouGisaia) [inline arlas-rest/src/main/java/io/arlas/server/rest/explore/aggregate/GeoAggregateRESTService.java:83]: It 's not an array, the object feature collection contains an attribute features that is a list

Comment (MohamedHamouGisaia) [reply arlas-rest/src/main/java/io/arlas/server/rest/explore/aggregate/GeoAggregateRESTService.java:83]: it applies for all geo-endpoints

File: arlas-rest/src/main/java/io/arlas/server/rest/explore/describe/DescribeCollectionRESTService.java (hunk 1 above)
Comment (alainbodiguel) [reply arlas-rest/src/main/java/io/arlas/server/rest/explore/describe/DescribeCollectionRESTService.java]: Done

File: arlas-rest/src/main/java/io/arlas/server/rest/explore/search/SearchRESTService.java (hunk 2 above)
Comment (alainbodiguel) [reply arlas-rest/src/main/java/io/arlas/server/rest/explore/search/SearchRESTService.java]: Done

File: arlas-rest/src/main/java/io/arlas/server/rest/explore/search/SearchRESTService.java (hunk 3 above)
Comment (alainbodiguel) [reply arlas-rest/src/main/java/io/arlas/server/rest/explore/search/SearchRESTService.java]: Done

File: arlas-rest/src/main/java/io/arlas/server/rest/explore/aggregate/GeoAggregateRESTService.java (hunk 4 above)
Comment (alainbodiguel) [reply arlas-rest/src/main/java/io/arlas/server/rest/explore/aggregate/GeoAggregateRESTService.java]: Done, please review that I did not forget any.

File: stac-api/src/main/java/io/arlas/server/stac/api/StacConformanceRESTService.java [hunk 5]
```diff
@@ -48,14 +50,18 @@ public StacConformanceRESTService(STACConfiguration configuration,
...
+            summary = "Conformance",
+            description = """
+                    The URIs of all conformance classes supported by the server.
+                    To support "generic" clients that want to access multiple OGC API Features implementations -
+                    and not "just" a specific API / server, the server declares the conformance classes it implements and conforms to.""")
     @ApiResponses(value = {
-            @ApiResponse(code = 200, message = "Successful operation", response = ConformanceClasses.class, responseContainer = "ConformanceClasses"),
-            @ApiResponse(code = 500, message = "Arlas Server Error.", response = Error.class)})
+            @ApiResponse(responseCode = "200", description = "Successful operation",
+                    content = @Content(array = @ArraySchema(schema = @Schema(implementation = ConformanceClasses.class)))),
```
Comment (MohamedHamouGisaia) [inline stac-api/src/main/java/io/arlas/server/stac/api/StacConformanceRESTService.java:61]: Returns `ConformanceClasses.class`(not an array)

File: openapi/openapi.json [hunk 6]
```diff
@@ -269,96 +428,119 @@
...
+            "type" : "integer",
+            "format" : "int32"
+          }
         } ],
         "responses" : {
           "200" : {
             "description" : "Successful operation",
-            "schema" : {
-              "$ref" : "#/definitions/AggregationResponse"
+            "content" : {
```
Comment (MohamedHamouGisaia) [inline openapi/openapi.json:515]: will be fixed by removing the @array annotation

File: openapi/openapi.json [hunk 7]
```diff
@@ -368,209 +550,285 @@
...
+        },
         "responses" : {
           "200" : {
             "description" : "Successful operation",
-            "schema" : {
-              "$ref" : "#/definitions/AggregationResponse"
+            "content" : {
+              "application/json;charset=utf-8" : {
+                "schema" : {
+                  "type" : "array",
```
Comment (MohamedHamouGisaia) [inline openapi/openapi.json:593]: will be fixed by removing the @array annotation

File: openapi/openapi.json [hunk 8]
```diff
@@ -992,91 +1326,116 @@
...
         } ],
         "responses" : {
           "200" : {
             "description" : "Successful operation",
-            "schema" : {
-              "$ref" : "#/definitions/ComputationResponse"
+            "content" : {
+              "application/json;charset=utf-8" : {
+                "schema" : {
+                  "type" : "array",
```
Comment (MohamedHamouGisaia) [inline openapi/openapi.json:1413]: will be fixed by removing the @array annotation

File: openapi/openapi.json [hunk 9]
```diff
@@ -1086,53 +1445,72 @@
...
+          }
+        },
         "responses" : {
           "200" : {
             "description" : "Successful operation",
-            "schema" : {
-              "$ref" : "#/definitions/ComputationResponse"
+            "content" : {
+              "application/json;charset=utf-8" : {
+                "schema" : {
```
Comment (MohamedHamouGisaia) [inline openapi/openapi.json:1487]: will be fixed by removing the @array annotation

File: openapi/openapi.json [hunk 10]
```diff
@@ -1144,79 +1522,100 @@
...
         } ],
         "responses" : {
           "200" : {
             "description" : "Successful operation",
-            "schema" : {
-              "$ref" : "#/definitions/Hits"
+            "content" : {
+              "application/json;charset=utf-8" : {
+                "schema" : {
+                  "type" : "array",
```
Comment (MohamedHamouGisaia) [inline openapi/openapi.json:1593]: will be fixed by removing the @array annotation

File: openapi/openapi.json [hunk 11]
```diff
@@ -1226,46 +1625,64 @@
...
+            }
+          }
+        },
         "responses" : {
           "200" : {
             "description" : "Successful operation",
-            "schema" : {
-              "$ref" : "#/definitions/Hits"
+            "content" : {
+              "application/json;charset=utf-8" : {
```
Comment (MohamedHamouGisaia) [inline openapi/openapi.json:1658]: will be fixed by removing the @array annotation

File: openapi/openapi.json [hunk 12]
```diff
@@ -1991,202 +2337,459 @@
...
         } ],
         "responses" : {
           "200" : {
             "description" : "Successful operation",
-            "schema" : {
-              "$ref" : "#/definitions/Hits"
+            "content" : {
+              "application/json;charset=utf-8" : {
+                "schema" : {
+                  "type" : "array",
```
Comment (MohamedHamouGisaia) [inline openapi/openapi.json:2749]: will be fixed by removing the @array annotation

File: openapi/openapi.json [hunk 13]
```diff
@@ -1991,202 +2337,459 @@
...
+                  "items" : {
+                    "$ref" : "#/components/schemas/Hits"
+                  }
+                }
+              },
+              "application/atom+xml" : {
```
Comment (MohamedHamouGisaia) [inline openapi/openapi.json:2755]: will be fixed by removing the @array annotation

File: stac-api/src/main/java/io/arlas/server/stac/api/StacSearchRESTService.java [hunk 14]
```diff
@@ -196,25 +199,28 @@ However, in cases where the box spans the antimeridian the first value (west-mos
...
+    @Operation(
+            summary = "Search STAC items with full-featured filtering.",
+            description = "Retrieve items matching filters. Intended as the standard, full-featured query API.")
     @ApiResponses(value = {
-            @ApiResponse(code = 200, message = "A feature collection.", response = StacFeatureCollection.class,
-                    responseContainer = "StacFeatureCollection"),
-            @ApiResponse(code = 400, message = "Invalid query parameter.", response = Error.class),
-            @ApiResponse(code = 500, message = "Arlas Server Error.", response = Error.class)})
+            @ApiResponse(responseCode = "200", description = "A feature collection.",
+                    content = @Content(array = @ArraySchema(schema = @Schema(implementation = StacFeatureCollection.class)))),
```
Comment (MohamedHamouGisaia) [inline stac-api/src/main/java/io/arlas/server/stac/api/StacSearchRESTService.java:208]: Returns `StacFeatureCollection.class` (not array)

File: stac-api/src/main/java/io/arlas/server/stac/api/StacSearchRESTService.java [hunk 15]
```diff
@@ -71,19 +71,21 @@ public StacSearchRESTService(STACConfiguration configuration,
...
+    @Operation(
+            summary = "Search STAC items with simple filtering.",
+            description = "Retrieve Items matching filters. Intended as a shorthand API for simple queries.")
     @ApiResponses(value = {
-            @ApiResponse(code = 200, message = "A feature collection.", response = StacFeatureCollection.class,
-                    responseContainer = "StacFeatureCollection"),
-            @ApiResponse(code = 400, message = "Invalid query parameter.", response = Error.class),
-            @ApiResponse(code = 500, message = "Arlas Server Error.", response = Error.class)})
+            @ApiResponse(responseCode = "200", description = "A feature collection.",
+                    content = @Content(array = @ArraySchema(schema = @Schema(implementation = StacFeatureCollection.class)))),
```
Comment (MohamedHamouGisaia) [inline stac-api/src/main/java/io/arlas/server/stac/api/StacSearchRESTService.java:79]: Returns StacFeatureCollection.class (not array)

File: stac-api/src/main/java/io/arlas/server/stac/api/StacRESTService.java [hunk 16]
```diff
@@ -271,6 +263,7 @@ protected StacFeatureCollection getStacFeatureCollection(CollectionReference col
             Arrays.asList("self", "next", "previous").forEach(rel -> {
                 if (context.containsKey(rel)) {
                     if (method.equals("POST")) {
+                        // TODO : fix when body is null!!
```
Comment (MohamedHamouGisaia) [inline stac-api/src/main/java/io/arlas/server/stac/api/StacRESTService.java:266]: needs to create an issue ?

File: stac-api/src/main/java/io/arlas/server/stac/api/StacCoreRESTService.java [hunk 17]
```diff
@@ -20,72 +20,70 @@
...
+                    Returns the root STAC Catalog or STAC Collection that is the entry point for users to browse with STAC Browser or for search engines to crawl.
+                    This can either return a single STAC Collection or more commonly a STAC catalog.
+                    The landing page provides links to the API definition (link relations `service-desc` and `service-doc`) and the STAC records such as collections/catalogs (link relation `child`) or items (link relation `item`).
+                    Extensions may add additional links with new relation types.""")
     @ApiResponses(value = {
-            @ApiResponse(code = 200, message = "The landing page provides links to the API definition (link relations " +
-                    "`service-desc` and `service-doc`) and the Feature Collection (path `/collections`, link relation " +
-                    "`data`).", response = LandingPage.class, responseContainer = "LandingPage"),
-            @ApiResponse(code = 500, message = "Arlas Server Error.", response = Error.class)})
+            @ApiResponse(responseCode = "200", description = """
```
Comment (MohamedHamouGisaia) [inline stac-api/src/main/java/io/arlas/server/stac/api/StacCoreRESTService.java:76]: returns LandingPage (not array)

File: stac-api/src/main/java/io/arlas/server/stac/api/StacCoreRESTService.java [hunk 18]
```diff
@@ -120,15 +118,12 @@ public Response getLandingPage(@Context UriInfo uriInfo,
...
     @GET
     @Produces("application/vnd.oai.openapi+json;version=3.0")
-    @ApiOperation(
-            value = "OpenAPI",
-            response = String.class)
+    @Operation(summary = "OpenAPI")
     @ApiResponses(value = {
-            @ApiResponse(code = 200, message = "OpenAPI specification", response = String.class, responseContainer = "OpenAPI")})
+            @ApiResponse(responseCode = "200", description = "OpenAPI specification",
+                    content = @Content(array = @ArraySchema(schema = @Schema(implementation = String.class))))
```
Comment (MohamedHamouGisaia) [inline stac-api/src/main/java/io/arlas/server/stac/api/StacCoreRESTService.java:124]: returns String (not array)

File: arlas-rest/src/main/java/io/arlas/server/rest/explore/count/CountRESTService.java [hunk 19]
```diff
@@ -138,37 +154,46 @@ public Response count(
...
-    @ApiOperation(value = "Count", produces = UTF8JSON, notes = "Count the number of elements found in the collection(s), given the filters", consumes = UTF8JSON, response = Hits.class)
-    @ApiResponses(value = {@ApiResponse(code = 200, message = "Successful operation", response = Hits.class, responseContainer = "ArlasHits"),
-            @ApiResponse(code = 500, message = "Arlas Server Error.", response = Error.class), @ApiResponse(code = 400, message = "Bad request.", response = Error.class)})
+    @Operation(
+            summary = "Count",
+            description = "Count the number of elements found in the collection(s), given the filters"
+    )
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successful operation",
+                    content = @Content(array = @ArraySchema(schema = @Schema(implementation = Hits.class)))),
```
[the remark this answers is not in the document]
Comment (alainbodiguel) [reply arlas-rest/src/main/java/io/arlas/server/rest/explore/count/CountRESTService.java:163]: ok

File: arlas-rest/src/main/java/io/arlas/server/rest/explore/search/SearchRESTService.java [hunk 20]
```diff
@@ -209,16 +219,25 @@ public Response search(
...
     @Produces(UTF8JSON)
     @Consumes(UTF8JSON)
-    @ApiOperation(value = "Search", produces = UTF8JSON, notes = Documentation.SEARCH_OPERATION, consumes = UTF8JSON, response = Hits.class)
-    @ApiResponses(value = {@ApiResponse(code = 200, message = "Successful operation", response = Hits.class, responseContainer = "ArlasHits"),
-            @ApiResponse(code = 500, message = "Arlas Server Error.", response = Error.class), @ApiResponse(code = 400, message = "Bad request.", response = Error.class)})
+    @Operation(
+            summary = "Search",
+            description = Documentation.SEARCH_OPERATION
+    )
+    @ApiResponses(value = {
```
[the remark this answers is not in the document]
Comment (alainbodiguel) [reply arlas-rest/src/main/java/io/arlas/server/rest/explore/search/SearchRESTService.java:226]: ok

File: mkDocs.sh [hunk 21]
```diff
@@ -16,9 +16,9 @@ mkdir -p target/tmp/typescript-fetch
 
 echo "=> Generate API"
 docker run --rm \
-    --mount dst=/input/api.json,src="$PWD/openapi/swagger.json",type=bind,ro \
+    --mount dst=/input/api.json,src="$PWD/openapi/openapi.json",type=bind,ro \
     --mount dst=/output,src="$PWD/target/tmp/typescript-fetch",type=bind \
-	gisaia/swagger-codegen-2.4.14 \
+	gisaia/swagger-codegen-3.0.42 \
```
[the remark this answers is not in the document]
Comment (alainbodiguel) [reply mkDocs.sh:21]: ok

File: stac-api/src/main/java/io/arlas/server/stac/api/StacCollectionsRESTService.java [hunk 22]
```diff
@@ -82,13 +83,15 @@ For each collection, a link to the items in the collection (path `/collections/{
                       * An optional title and description for the collection;
                       * An optional extent that can be used to provide an indication of the spatial and temporal extent of the collection - typically derived from the data;
                       * An optional indicator about the type of the items in the collection (the default value, if the indicator is not provided, is 'feature').""",
-                    response = CollectionList.class, responseContainer = "CollectionList"),
-            @ApiResponse(code = 500, message = "Arlas Server Error.", response = Error.class)})
+                    content = @Content(array = @ArraySchema(schema = @Schema(implementation = CollectionList.class)))),
```
[the remark this answers is not in the document]
Comment (alainbodiguel) [reply stac-api/src/main/java/io/arlas/server/stac/api/StacCollectionsRESTService.java:86]: ok

File: stac-api/src/main/java/io/arlas/server/stac/api/StacSearchRESTService.java (hunk 15 above)
Comment (alainbodiguel) [reply stac-api/src/main/java/io/arlas/server/stac/api/StacSearchRESTService.java:79]: ok

File: stac-api/src/main/java/io/arlas/server/stac/api/StacRESTService.java (hunk 16 above)
Comment (alainbodiguel) [reply stac-api/src/main/java/io/arlas/server/stac/api/StacRESTService.java:266]: https://github.com/gisaia/ARLAS-server/issues/948

File: stac-api/src/main/java/io/arlas/server/stac/api/StacCoreRESTService.java (hunk 17 above)
Comment (alainbodiguel) [reply stac-api/src/main/java/io/arlas/server/stac/api/StacCoreRESTService.java:76]: ok

File: stac-api/src/main/java/io/arlas/server/stac/api/StacCoreRESTService.java (hunk 18 above)
Comment (alainbodiguel) [reply stac-api/src/main/java/io/arlas/server/stac/api/StacCoreRESTService.java:124]: ok

File: openapi/openapi.json (hunk 6 above)
Comment (alainbodiguel) [reply openapi/openapi.json:515]: ok

File: openapi/openapi.json (hunk 7 above)
Comment (alainbodiguel) [reply openapi/openapi.json:593]: ok

File: openapi/openapi.json (hunk 8 above)
Comment (alainbodiguel) [reply openapi/openapi.json:1413]: ok

File: openapi/openapi.json (hunk 10 above)
Comment (alainbodiguel) [reply openapi/openapi.json:1593]: ok

File: openapi/openapi.json (hunk 11 above)
Comment (alainbodiguel) [reply openapi/openapi.json:1658]: ok

File: openapi/openapi.json (hunk 12 above)
Comment (alainbodiguel) [reply openapi/openapi.json:2749]: ok

File: openapi/openapi.json (hunk 13 above)
Comment (alainbodiguel) [reply openapi/openapi.json:2755]: ok

File: openapi/openapi.json [hunk 23]
```diff
@@ -2196,53 +2799,72 @@
...
+              }
+            }
+          }
+        },
         "responses" : {
           "200" : {
             "description" : "Successful operation",
-            "schema" : {
-              "$ref" : "#/definitions/Hits"
+            "content" : {
```
[the remark this answers is not in the document]
Comment (alainbodiguel) [reply openapi/openapi.json:2839]: ok

File: stac-api/src/main/java/io/arlas/server/stac/api/StacSearchRESTService.java (hunk 14 above)
Comment (alainbodiguel) [reply stac-api/src/main/java/io/arlas/server/stac/api/StacSearchRESTService.java:208]: ok

File: stac-api/src/main/java/io/arlas/server/stac/api/StacConformanceRESTService.java (hunk 5 above)
Comment (alainbodiguel) [reply stac-api/src/main/java/io/arlas/server/stac/api/StacConformanceRESTService.java:61]: ok

File: arlas-rest/src/main/java/io/arlas/server/rest/explore/compute/ComputeRESTService.java [hunk 24]
```diff
@@ -55,79 +60,88 @@ public ComputeRESTService(ExploreService exploreService) {
...
-    @ApiResponses(value = {@ApiResponse(code = 200, message = "Successful operation", response = ComputationResponse.class, responseContainer = "ArlasComputation"),
-            @ApiResponse(code = 500, message = "Arlas Server Error.", response = Error.class),
-            @ApiResponse(code = 400, message = "Bad request.", response = Error.class)})
+    @Operation(
+            summary = "Compute",
+            description = Documentation.COMPUTE_OPERATION
+    )
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successful operation",
+                    content = @Content(array = @ArraySchema(schema = @Schema(implementation = ComputationResponse.class)))),
```
[the remark this answers is not in the document]
Comment (MohamedHamouGisaia) [reply arlas-rest/src/main/java/io/arlas/server/rest/explore/compute/ComputeRESTService.java:69]: @alainbodiguel forgotten array :p , it should be ComputationResponse

File: arlas-rest/src/main/java/io/arlas/server/rest/explore/count/CountRESTService.java [hunk 25]
```diff
@@ -54,60 +59,71 @@ public CountRESTService(ExploreService exploreService) {
...
```
[the remark this answers is not in the document]
Comment (MohamedHamouGisaia) [reply arlas-rest/src/main/java/io/arlas/server/rest/explore/count/CountRESTService.java:68]: @alainbodiguel forgotten array :p , it should be Hits.class

File: arlas-rest/src/main/java/io/arlas/server/rest/explore/search/SearchRESTService.java [hunk 26]
```diff
@@ -59,121 +64,126 @@ public SearchRESTService(ExploreService exploreService) {
...
-    @ApiOperation(value = "Search", produces = UTF8JSON + "," + ATOM.APPLICATION_ATOM_XML, notes = Documentation.SEARCH_OPERATION, consumes = UTF8JSON, response = Hits.class)
-    @ApiResponses(value = {@ApiResponse(code = 200, message = "Successful operation", response = Hits.class, responseContainer = "ArlasHits"),
-            @ApiResponse(code = 500, message = "Arlas Server Error.", response = Error.class), @ApiResponse(code = 400, message = "Bad request.", response = Error.class)})
+    @Operation(
+            summary = "Search",
+            description = Documentation.SEARCH_OPERATION
+    )
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successful operation",
+                    content = @Content(array = @ArraySchema(schema = @Schema(implementation = Hits.class)))),
```
[the remark this answers is not in the document]
Comment (MohamedHamouGisaia) [reply arlas-rest/src/main/java/io/arlas/server/rest/explore/search/SearchRESTService.java:73]: @alainbodiguel forgotten array :p , it should be Hits.class

File: arlas-rest/src/main/java/io/arlas/server/rest/explore/compute/ComputeRESTService.java (hunk 24 above)
[the remark this answers is not in the document]
Comment (alainbodiguel) [reply arlas-rest/src/main/java/io/arlas/server/rest/explore/compute/ComputeRESTService.java:69]: ok

File: arlas-rest/src/main/java/io/arlas/server/rest/explore/count/CountRESTService.java (hunk 25 above)
[the remark this answers is not in the document]
Comment (alainbodiguel) [reply arlas-rest/src/main/java/io/arlas/server/rest/explore/count/CountRESTService.java:68]: ok

**Review by mbarbet — APPROVED** [review]

**Pull request merged**